packages/react-devtools-shared/src/__tests__/profilerChangeDescriptions-test.js

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import typeof ReactTestRenderer from 'react-test-renderer';
+
+describe('Profiler change descriptions', () => {
+  let React;
+  let TestRenderer: ReactTestRenderer;
+  let legacyRender;
+  let store: Store;
+  let utils;
+
+  beforeEach(() => {
+    utils = require('./utils');
+    utils.beforeEachProfiling();
+
+    legacyRender = utils.legacyRender;
+
+    store = global.store;
+    store.collapseNodesByDefault = false;
+    store.recordChangeDescriptions = true;
+
+    React = require('react');
+    TestRenderer = utils.requireTestRenderer();
+  });
+
+  it('should identify when context is the reason for an udpate', () => {
+    const Context = React.createContext(0);
+
+    function Child() {
+      const context = React.useContext(Context);
+      return context;
+    }
+
+    const MemoizedChild = React.memo(Child);
+    const ForwardRefChild = React.forwardRef(function ForwardRef(props, ref) {
+      return <Child />;
+    });
+
+    let forceUpdate = null;
+
+    const MemoizedApp = React.memo(function App() {
+      const [val, dispatch] = React.useReducer(x => x + 1, 0);
+
+      forceUpdate = dispatch;
+
+      return (
+        <Context.Provider value={val}>
+          <Child />
+          <MemoizedChild />
+          <ForwardRefChild />
+        </Context.Provider>
+      );
+    });
+
+    const container = document.createElement('div');
+
+    utils.act(() => store.profilerStore.startProfiling());
+    utils.act(() => legacyRender(<MemoizedApp />, container));
+    utils.act(() => forceUpdate());
+    utils.act(() => store.profilerStore.stopProfiling());
+
+    const rootID = store.roots[0];
+
+    let commitData = null;
+
+    function Validator({commitIndex}) {
+      commitData = store.profilerStore.getCommitData(rootID, commitIndex);
+      return null;
+    }
+
+    utils.act(() => {
+      TestRenderer.create(<Validator commitIndex={1} rootID={rootID} />);
+    });
```

Comment:
Why does this need to be inside a component instead of just accessing `store.profilerStore.getCommitData` inline?

Reply:
It's not necessary. I'm just used to writing tests for stuff that _suspends_. I'll fix it before the PR is moved out of _Draft_ status.